ontology/obo/tree.go

```diff
@@ -0,0 +1,160 @@
+package obo
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+// BuildTree takes a slice of Obo structs and builds a tree, setting
+// the Parents and Children fields to point to related nodes.
+func BuildTree(terms []Obo) map[string]*Obo {
+	termMap := makeTermMap(terms)
+
+	var parentId string
+	var foundInMap bool
+	var parentTerm *Obo
+	var isAItem IsADescription
+
+	// Now we iterate through terms and set up the tree
+	for i := range terms {
+		term := &terms[i] //get a memory copy
+		for _, isAItem = range term.IsA {
+			parentId = isAItem.ParentId
+			if parentTerm, foundInMap = termMap[parentId]; foundInMap {
+				term.Parents = append(term.Parents, parentTerm)
+				parentTerm.Children = append(parentTerm.Children, term)
+			} else {
+				//log.Fatalf("Error: The term with ID \"%s\" has a parent with ID \"%s\" that "+
```

Comment:
I think this should be a fatal.  If they say a node exists, that does not exist, that seems like an error

Reply:
Next commit adds the "force" option to BuildTree, allowing us to control this behavior

Reply:
Is the "force" option needed for the Gene Ontology files?  It is a good option to have if there are bugs in the GO files, but I am worried we are giving researchers wiggle room that they do not need, which may lead to silent errors.